app/scripts/views/visits/index.js

```diff
@@ -78,16 +80,24 @@ define([
       }
 
       // fire off the xhr request
-      var xhr = this.collection.fetch({ reset: true, data: data });
+      this.collection.fetch({ reset: true, data: data })
+        .done(this._fetchDone.bind(this))
+        .fail(this._fetchFailed.bind(this));
+    },
 
-      // keep track of the state of things
-      xhr.done(function (data) {
-        this.loadingMoreVisits = false;
+    _fetchDone: function (data) {
+      this.loadingMoreVisits = false;
 
-        // if the length is less than the count we know there aren't more results
-        this.hasMoreVisits = this.collection.length === this.count;
-        this.lastVisitId = this.collection.last().get('id');
-      }.bind(this));
+      // if the length is less than the count we know there aren't more results
+      this.hasMoreVisits = this.collection.length === this.count;
+      this.lastVisitId = this.collection.last().get('id');
+    },
+
+    _fetchFailed: function (xhr) {
+      // 404 means no results. keep retrying every NO_RESULTS_RETRY_WAIT ms
+      if (xhr.status === 404) {
+        this.fetchRetryTimer = setTimeout(this._fetch.bind(this), this.FETCH_RETRY_WAIT);
```

Comment:
Yeah I put this in here so we could experiment with what it feels like to have data loaded when the welcome experience ends. The reality is that we're going to want to do something rather fancy, and this very likely won't be the right way to do it. I'll just remove this for now.